### PR TITLE
Enable `custom-property-disallowed-list` in `stylelint-polaris`

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?

Follow up to #8968

### WHAT is this pull request doing?

Adds the `custom-property-disallowed-list` plugin to the `stylelint-polaris` config to disallow a collection of Polaris v10 custom properties prior to the v11 release.

### CODE REVIEW

File: .changeset/khaki-walls-wash.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@shopify/polaris': minor
+'polaris.shopify.com': minor
+'@shopify/stylelint-polaris': minor
+---
+
+Enable `custom-property-disallowed-list` rule
```
Comment (chloerice) [inline .changeset/khaki-walls-wash.md]:
```suggestion
Enabled the `custom-property-disallowed-list` rule
```

**Review by chloerice — APPROVED** [review]
This looks good to me 💯 Love how the new rule allows these properties to be categorized 🙌🏽

**Review by alex-page — APPROVED** [review]

File: polaris.shopify.com/content/tools/stylelint-polaris/rules/layout-custom-property-disallowed-list.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+title: layout/custom-property-disallowed-list
+description: Disallows use of layout legacy custom properties.
```
Comment (lgriffee) [inline polaris.shopify.com/content/tools/stylelint-polaris/rules/layout-custom-property-disallowed-list.md]:
```suggestion
description: Disallows use of legacy layout custom properties.
```

File: .changeset/khaki-walls-wash.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+'@shopify/stylelint-polaris': major
+'@shopify/polaris': minor
+'polaris.shopify.com': minor
+---
+
+Add deprecated v10 custom properties to `stylelint-polaris` `global-disallowed-list` and enabled the `custom-property-disallowed-list` rule.
```
Comment (lgriffee) [inline .changeset/khaki-walls-wash.md]:
```suggestion
Enabled the `custom-property-disallowed-list` rule and added deprecated v10 custom properties.
```

**Review by aaronccasanova — APPROVED** [review]

**Pull request merged**